underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -188,7 +188,7 @@ public boolean delete(String path, boolean recursive) throws IOException {
         LOG.error("Unable to delete {} because listInternal returns null", path);
         return false;
       }
-      if (isFolder(path) && children.length != 0) {
+      if (isDirectory(path) && children.length != 0) {
```

Comment:
valid point. i'll address this when abstracting out object stores. the same code is repeated in some other object stores.